[torch.nn Onboarding Lab] adding a new module 'Bias'

NivekT opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #62992

This PR is created as an onboarding exercise. This will not be merged with master.

NivekT committed:
torch.nn onboarding lab initial commit

[ghstack-poisoned]

NivekT committed:
Update on "[torch.nn Onboarding Lab] adding a new module 'Bias'"


This PR is created as an onboarding exercise. This will not be merged with master.

[ghstack-poisoned]

NivekT committed:
Update on "[torch.nn Onboarding Lab] adding a new module 'Bias'"


This PR is created as an onboarding exercise. This will not be merged with master.

[ghstack-poisoned]

NivekT committed:
Update on "[torch.nn Onboarding Lab] adding a new module 'Bias'"


This PR is created as an onboarding exercise. This will not be merged with master.

[ghstack-poisoned]

jbschlosser reviewed (commented):
Nice work!! just a couple small comments below but the implementation looks correct overall :)

NivekT commented on test/test_nn.py:
```diff
@@ -17601,6 +17601,15 @@ def test_lazy_share_memory_buffer(self):
+        module = nn.Bias(5)
+        input = torch.ones(10, 5)
+        module(input)
+        self.assertIsInstance(module, nn.Bias)
```
Removed.

NivekT commented on test/test_nn.py:
```diff
@@ -17601,6 +17601,15 @@ def test_lazy_share_memory_buffer(self):
+        self.assertIsInstance(module, nn.Bias)
+        self.assertTrue(module.bias_values.shape == (5,))
+        y = module(input)
+        self.assertTrue(torch.equal(torch.nn.functional.bias(input, module.bias_values), y))
```
Switched to `self.assertEqual()` and changing `y` to `y = input + module.bias_values`

NivekT commented on torch/csrc/api/include/torch/nn/functional/linear.h:
```diff
@@ -26,6 +26,12 @@ inline Tensor linear(const Tensor& input, const Tensor& weight,
+// ============================================================================
+
+inline Tensor bias(const Tensor& input, const Tensor& bias_values) {
+  return input + bias_values;
```
Switched to calling core operator logic.

NivekT commented on torch/nn/functional.py:
```diff
@@ -1827,6 +1827,17 @@ def hardsigmoid(input: Tensor, inplace: bool = False) -> Tensor:
+    of the incoming data :math:`y = x + b`
+
+    Shape:
+        - Input: :math:`(*, H_{n})` where :math:`*` means any number of
```
Yes, that shouldn't have any problem. I think the wording should be "including None"

NivekT commented on torch/nn/functional.py:
```diff
@@ -1827,6 +1827,17 @@ def hardsigmoid(input: Tensor, inplace: bool = False) -> Tensor:
+
+    Shape:
+        - Input: :math:`(*, H_{n})` where :math:`*` means any number of
+          dimensions excluding none and :math:`H_{n} = \text{n\_features}`.
```
Fixed.

NivekT commented on torch/testing/_internal/common_nn.py:
```diff
@@ -98,6 +98,15 @@ def get_weight(m):
 # - `has_parity`: if `False`, expects this test dict to fail the C++ parity test. Default: True.
 
 module_tests = [
+    dict(
```
I see, thanks for pointing that out.

NivekT committed:
Update on "[torch.nn Onboarding Lab] adding a new module 'Bias'"


This PR is created as an onboarding exercise. This will not be merged with master.

[ghstack-poisoned]

jbschlosser reviewed (commented):
LGTM!

NivekT commented:
This exercise PR looks good and can be closed.
